src/systems/user_commands/UserCommands.cc

```diff
@@ -683,6 +798,175 @@ bool PoseCommand::Execute()
+  {
+    ignwarn << "Can't create component that already exists" << std::endl;
+    return false;
+  }
+
+  this->iface->ecm->
+    CreateComponent(entityMsg->id(), components::ContactSensorData());
+  igndbg << "Enabled collision [" << entityMsg->id() << "]" << std::endl;
+
+  return true;
+}
+
+//////////////////////////////////////////////////
+bool EnableCollisionCommand::HasContactSensor(const Entity _collision)
+{
+  auto *ecm = this->iface->ecm;
+
+  auto *linkEntity = ecm->Component<components::ParentEntity>(_collision);
+  auto allLinkSensors =
+    ecm->EntitiesByComponents(components::Sensor(),
+      components::ParentEntity(*linkEntity));
+
+  for (auto const &sensor : allLinkSensors)
+  {
+    // Check if it is a contact sensor
+    auto isContactSensor =
+      ecm->EntityHasComponentType(sensor, components::ContactSensor::typeId);
+    if (!isContactSensor)
+      continue;
+
+    // Check if sensor is connected to _collision
+    auto componentName = ecm->Component<components::Name>(_collision);
+    std::string collisionName = componentName->Data();
+    auto sensorSDF = ecm->Component<components::ContactSensor>(sensor)->Data();
+    auto sensorCollisionName =
+      sensorSDF->GetElement("contact")->Get<std::string>("collision");
+
+    if (collisionName == sensorCollisionName)
+    {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+//////////////////////////////////////////////////
+DisableCollisionCommand::DisableCollisionCommand(msgs::Entity *_msg,
+    std::shared_ptr<UserCommandsInterface> &_iface)
+    : UserCommandBase(_msg, _iface)
+{
+}
+
+//////////////////////////////////////////////////
+bool DisableCollisionCommand::Execute()
+{
+  auto entityMsg = dynamic_cast<const msgs::Entity *>(this->msg);
+  if (nullptr == entityMsg)
+  {
+    ignerr << "Internal error, null create message" << std::endl;
+    return false;
+  }
+
+  // Check Entity type
+  if (entityMsg->type() != msgs::Entity::COLLISION)
+  {
+    ignwarn << "Expected msgs::Entity::Type::COLLISION, exiting service..."
+      << std::endl;
+    return false;
+  }
+
+  // Check if collision is connected to a contact sensor
+  if (this->HasContactSensor(entityMsg->id()))
+  {
+    ignwarn << "Requested collision is connected to a contact sensor, "
+      << "exiting service..." << std::endl;
+    return false;
+  }
+
+  // Remove ContactSensorData component
+  auto *contactDataComp =
+    this->iface->ecm->Component<
+      components::ContactSensorData>(entityMsg->id());
+  if (!contactDataComp)
+  {
+    ignwarn << "No ContactSensorData detected inside entity " << entityMsg->id()
+      << std::endl;
+    return false;
+  }
+
+  this->iface->ecm->
+    RemoveComponent(entityMsg->id(), components::ContactSensorData::typeId);
+
+  igndbg << "Disabled collision [" << entityMsg->id() << "]" << std::endl;
+
+  return true;
+}
+
+//////////////////////////////////////////////////
+bool DisableCollisionCommand::HasContactSensor(const Entity _collision)
```

Comment:
The `DisableCollisionCommand::HasContactSensor` and `EnableCollisionCommand::HasContactSensor` functions look duplicate. It would be nice to reuse the same function so that if we change one of them, we change both at the same time. Maybe it could be a function in `UserCommandsInterface`?

Reply:
Yep, they are duplicate. I just wasn't sure where to put that, thanks for the suggestion ;)